src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -492,24 +487,29 @@ public void restoreFeedbackSessionFromRecycleBin(String feedbackSessionName, Str
      * Gets the expected number of submissions for a feedback session.
      */
     public int getExpectedTotalSubmission(FeedbackSessionAttributes fsa) {
-        List<StudentAttributes> students = studentsLogic.getStudentsForCourse(fsa.getCourseId());
-        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(fsa.getCourseId());
-        List<FeedbackQuestionAttributes> questions =
-                fqLogic.getFeedbackQuestionsForSession(fsa.getFeedbackSessionName(), fsa.getCourseId());
-        List<FeedbackQuestionAttributes> studentQns = fqLogic.getFeedbackQuestionsForStudents(questions);
+        int numOfStudents = studentsLogic.getNumberOfStudentsForCourse(fsa.getCourseId());
+        List<String> instructorEmails = instructorsLogic.getInstructorEmailsForCourse(fsa.getCourseId());
```

Comment:
These values are not needed if the session has no question for the relevant user type. Eagerly fetching them is a wasted operation.